### PR TITLE
[Merged by Bors] - feat: lower bound on the Ruzsa-Szemerédi problem

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanCamCombi

---

- [x] depends on: #19505

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/Extremal/RuzsaSzemeredi.lean [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+  haveI := noAccidental hs; TripartiteFromTriangles.locallyLinear _
+
+private lemma card_edgeFinset (hs : ThreeAPFree (s : Set α)) [DecidableEq α] :
+    #(graph <| triangleIndices s).edgeFinset = 3 * card α * #s := by
+  haveI := noAccidental hs
+  rw [(locallyLinear hs).card_edgeFinset, card_triangles, card_triangleIndices, mul_assoc]
+
+end RuzsaSzemeredi
+
+variable (α) [Fintype α] [DecidableEq α] [CommRing α] [Fact <| IsUnit (2 : α)]
```
Comment (b-mehta) [inline Mathlib/Combinatorics/Extremal/RuzsaSzemeredi.lean:167]: Is `Invertible 2` preferred here?

**Review by b-mehta — COMMENTED** [review]
Nice!!

bors d+

File: Mathlib/Combinatorics/Extremal/RuzsaSzemeredi.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Combinatorics/Extremal/RuzsaSzemeredi.lean:167]: No because it's unnecessary data

**Comment by YaelDillies** [comment]
Finally! This was written a long time ago now

bors merge

**Pull request closed without merging**